querier.hh

```diff
@@ -180,6 +200,24 @@ public:
         }
         return full_position_view(dk->key(), _compaction_state->current_position());
     }
+
+private:
+    sstring current_range_to_string() const {
+        sstring keystr = "<no key>";
+        std::optional<partition_key> key(std::nullopt);
+        if (const dht::partition_range *prange = _range.get()) {
+            if (const auto &start = prange->start()) {
+                key = start->value().key();
+            } else if (const auto &end = prange->end()) {
+                key = end->value().key();
+            }
+        }
+        if (key) {
+            keystr = format("{}", (*key).with_schema(*_schema.get()));
+            return utils::utf8::validate((const uint8_t *) keystr.data(), keystr.size()) ? keystr : "<non-utf8-key>";
```

Comment:
We have a similar method in https://github.com/scylladb/scylladb/blob/78850884d22fc4f84a13c683f2eaacc6a0e6b3bf/compaction/compaction.cc#L1337-L1340. I think this should be integrated directly into https://github.com/scylladb/scylladb/blob/78850884d22fc4f84a13c683f2eaacc6a0e6b3bf/keys.cc#L23-L36, instead of having many reimplementations in the code-base.

Reply:
You mean, to move partition_key_to_string() function from compaction.cc into the class partition_key, as public method?

Reply:
Actually, I think we should move it to `operator<<(partition_key::with_schema_wrapper)`.